BUILD.bazel

```diff
@@ -15,6 +15,25 @@ buildifier_exclude_patterns = [
     "./vendor/**",
 ]
 
+filegroup(
+    name = "all_files",
+    testonly = True,
+    srcs = [
+        "BUILD.bazel",
+        "WORKSPACE",
+        "constants.bzl",
+        "//debug/linking_utils:all_files",
+        "//docs:all_files",
+        "//haskell:all_files",
+        "//nixpkgs:all_files",
+        "//protobuf:all_files",
+        "//rule_info:all_files",
+        "//tests:all_files",
```

Comment:
For the buildifer use-case we'll want to include all files under `tests` to make sure that they are format tested as well. But, for the `go_bazel_test` use-case perhaps not. Including them means that `go_bazel_test` type integration tests are re-run not only when rules_haskell's implementation changes, but also when any of its tests changes.
AFAIK rules_go does not include the tests tree in the `all_files` filegroups.
A simple option might be to omit `//tests:all_files` here and then add it to the buildifier test explicitly. What do you think?

Reply:
A related instance is `//vendor`. It's needed for functionality, so will be needed for `go_bazel_test`, but it would be preferable to skip it for buildifier tests.

Reply:
Probably it  makes sense to enumerate all necessary filegroups in ```buildifier_test``` srcs explicitly, because otherwise we'll inevitably end up with some kind of "sources for buildifer_test" filegroup. In this case name ```all_files``` is a bit odd  for filegroup dedicated for integration_tests. ```rules_go``` for example doesn't include tests in ```all_files``` but use it both for integration tests and buildifier which is a bit kludgy imho. Some bazelbuild projects like ```bazel-skylib``` or ```rules_python``` uses name ```distribution``` for such filegroup, probably it makes sense to follow

Reply:
> Some bazelbuild projects like `bazel-skylib` or `rules_python` uses name `distribution` for such filegroup, probably it makes sense to follow

Nice find! Yes, that sounds like a good precedent to follow.

> Probably it makes sense to enumerate all necessary filegroups in `buildifier_test` srcs explicitly, [...]

You mean at the top-level something like this?
```
buildifier_test(
    srcs = [
        "//haskell:all_files",
        "//nixpkgs:all_files",
        "//tests:all_files",
        ...
    ],
```
Yes, that sounds like a good solution to the problem.

Reply:
> You mean at the top-level something like this?

Yes, that's exactly what I meant :)

Reply:
> Some bazelbuild projects like bazel-skylib or rules_python uses name distribution for such filegroup, probably it makes sense to follow